### ISSUE TITLE
Script eval() disallowed when a Content Security Policy header is applied

### ISSUE THREAD

**Issue by russback** [verbb/formie#336]
**Description**
Console error: 
```
Uncaught EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".

    at HTMLScriptElement.registeredJs.length.registeredJs.forEach.config.src.$script.onload (formie.js?v=1612422496:1)
```

Can the eval() be removed in favour of something safer?

**Steps to reproduce**
1. Set a Content Security Policy header of `script-src 'self'`
2. Visit a page with Formie output.

**Additional info**
- Plugin version: 1.3.16.1
- Craft version: 3.6.3
- Multi-site: Yes

**Comment by engram-design** [comment]
I thought this might come up one day. The reason behind using `eval()` to to allow custom fields to run their own JS functions. In this scenario, it's safe in that it's never dealing with user modified content. The JS initialisation code is delivered from server-side code associated with the field and evaluated.

Basically, it takes a string defined by the field class - let's say `FormieFileUpload`, and creates a new JS class, passing in params, essentially `eval(`new ${config.module}(fieldConfig)`);` which equates to `new FormieFileUpload({})`

As such, I believe this is quite safe. So far, I haven't been able to figure out a good way to handle this dynamic need.

I can have a look at [safe-eval](https://www.npmjs.com/package/safe-eval) to see if that'll alleviate warnings.

**Comment by engram-design** [comment]
Well, I'm not sure if it's just fresh eyes, but looks like this can be achieved fairly easily. Should be fixed in the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.3.16.1",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Should be fixed in [1.3.17](https://github.com/verbb/formie/releases/tag/1.3.17)
